### PR TITLE
Add "complete action" support for missing actions

### LINKED ISSUE #1826: title not showing after action creation
## Steps to reproduce

1. Create an action (I've been able to replicate this with manage permissions / edit team)
2. Fill in the fields
3. submit the form

## Expected behaviour

* All of the form fields are displayed correctly as to what the user had entered

## Actual behaviour

* The title of the action is displayed as empty. if the page is refreshed the title of the action appears however this shouldn't be necessary
* This is occurring in multiple action types
Screenshot 2024-01-31 at 10 29 33

### PR DESCRIPTION
**Pull request by ArmandoGraterol**
## Description

This PR supports the actions missing in the "Completed action" version of the action sidebar. This includes changes to most of the actions but mainly: create/edit domains, manage colony objective, and edit colony details.

Completed "Manage permissions" action:
Screenshot 2024-02-19 at 17 01 30

## Testing

* Make sure to test each action and its motion version. We have to make sure that the titles and data in each of the sidebars are displayed correctly.
* The only action not covered here is `Manage tokens` as that one still has a PR up with conflicting changes to it: https://github.com/JoinColony/colonyCDapp/pull/1430

Resolves #1826, #1644, #1646, #1651

### CODE REVIEW

**Review by bassgeta — CHANGES_REQUESTED** [review]
Just checked the code for now, looks really great, nice job unifying the rows :D! 
dropped a few comments which are minor. 
I just have another question/proposal. Should we unify these title strings to use the ones from `en-actions.ts` like we use em [here](https://github.com/JoinColony/colonyCDapp/blob/master/src/components/v5/common/ActionSidebar/partials/ActionSidebarDescription/partials/CreateDecisionDescription.tsx) ?

File: src/components/v5/common/CompletedAction/partials/rows/TeamFrom.tsx [hunk 1]
```diff
@@ -1,39 +1,50 @@
...
+const TeamFromRow = ({ teamMetadata, actionType }: TeamFromRowProps) => {
+  const getTooltipContent = () => {
+    switch (actionType) {
+      case ColonyActionType.SetUserRoles:
+        return formatText({
+          id: 'actionSidebar.tooltip.managePermissions.team',
+        });
+      default:
+        return formatText({
+          id: 'actionSidebar.tooltip.simplePayment.from',
```
[the remark this answers is not in the document]
Comment (ArmandoGraterol) [reply src/components/v5/common/CompletedAction/partials/rows/TeamFrom.tsx:26]: What's the typo? :ˆ)

File: src/components/v5/common/CompletedAction/partials/UpgradeColonyObjective/UpgradeColonyObjective.tsx [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+  ActionTitle,
+} from '../Blocks/index.ts';
+import {
+  ActionTypeRow,
+  CreatedInRow,
+  DecisionMethodRow,
+  DescriptionRow,
+} from '../rows/index.ts';
+
+const displayName = 'v5.common.CompletedAction.partials.EditColonyDetails';
```
[the remark this answers is not in the document]
Comment (ArmandoGraterol) [reply src/components/v5/common/CompletedAction/partials/UpgradeColonyObjective/UpgradeColonyObjective.tsx:22]: Serves me right for being lazy and copy-pasting stuff :)

File: src/components/v5/common/CompletedAction/partials/rows/PermissionsTable.tsx [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+  domainId: number | undefined;
+  userColonyRoles: ColonyRole[];
+}
+
+const PermissionsTable = ({ role, domainId, userColonyRoles }: Props) => {
+  const customPermissionsTableColumns =
+    getCustomPermissionsTableColumns(userColonyRoles);
+  const permissionsTableProps = usePermissionsTableProps(role);
+
+  if (!role) {
```
[the remark this answers is not in the document]
Comment (ArmandoGraterol) [reply src/components/v5/common/CompletedAction/partials/rows/PermissionsTable.tsx:29]: True! It's probably a leftover from when I was adapting the form permission table

File: src/components/v5/common/CompletedAction/partials/rows/hooks.tsx [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+import React from 'react';
+
+import { type CustomPermissionTableModel } from '~types/permissions.ts';
+import { formatText } from '~utils/intl.ts';
+import Switch from '~v5/common/Fields/Switch/index.ts';
+
+const customPermissionsColumnHelper =
+  createColumnHelper<CustomPermissionTableModel>();
+
+export const getCustomPermissionsTableColumns = (
```
[the remark this answers is not in the document]
Comment (ArmandoGraterol) [reply src/components/v5/common/CompletedAction/partials/rows/hooks.tsx:12]: I don't mind changing the name of this file to something else as there are no hooks being used here anymore but why `tableFixtures`? (I had to search what `fixtures` is :P) as opposed to something like `consts` or `utils`?

File: src/components/v5/common/CompletedAction/partials/rows/TeamFrom.tsx (hunk 1 above)
[the remark this answers is not in the document]
Comment (bassgeta) [reply src/components/v5/common/CompletedAction/partials/rows/TeamFrom.tsx:26]: ah you're right this one is intended, I thought we need a generic from, resolving this comment

File: src/components/v5/common/CompletedAction/partials/EditColonyDetails/EditColonyDetails.tsx [hunk 5]
```diff
@@ -90,6 +107,7 @@ const EditColonyDetails = ({ action }: EditColonyDetailsProps) => {
       {action.annotation?.message && (
         <DescriptionRow description={action.annotation.message} />
       )}
+      <SocialLinksTable socialLinks={actionColonyMetadata?.externalLinks} />
```
[the remark this answers is not in the document]
Comment (bassgeta) [reply src/components/v5/common/CompletedAction/partials/EditColonyDetails/EditColonyDetails.tsx:110]: yeah maybe we just not render the component if we have undefined data and the component can always operate with the fact that data is defined :+1:

File: src/components/v5/common/CompletedAction/partials/rows/hooks.tsx (hunk 4 above)
[the remark this answers is not in the document]
Comment (bassgeta) [reply src/components/v5/common/CompletedAction/partials/rows/hooks.tsx:12]: I used fixtures due to me naming it so in another project, but I am not opinionated on the naming (i suck at naming after all). so we can rename it to a utils.tsx for the moment or whatever you feel is best

**Review by bassgeta — APPROVED** [review]
all good now, great job, tested out some actions and works like a charm :+1:

File: src/components/v5/common/ActionSidebar/partials/forms/ManagePermissionsForm/consts.tsx [hunk 6]
```diff
@@ -29,7 +29,7 @@ export const validationSchema = object()
   .shape({
     member: string().required(),
     team: number().required(),
-    createdIn: number(),
+    createdIn: number().defined(),
```
Comment (iamsamgibbs) [inline src/components/v5/common/ActionSidebar/partials/forms/ManagePermissionsForm/consts.tsx:32]: I think this change is preventing the form from being submitted when using the permissions decision method.

Screenshot 2024-02-22 at 10 45 44

**Review by iamsamgibbs — CHANGES_REQUESTED** [review]
Great work on this! All the code changes look great and working as expected.

I've left one code comment where a validation change is preventing the manage permissions form from submitting with permissions.

The completed view for "manage colony objective" is also missing the percentage sign after the number.

Screenshot 2024-02-22 at 10 26 32

(Unrelated to this issue, but I also noticed the "edit team" action is missing the description field, but it looks like the completed action view would show this correctly if it were there - not sure if that is quick and easy for you to add in with this PR, or can open as a new issue)

Screenshot 2024-02-22 at 10 24 37

**Comment by ArmandoGraterol** [comment]
@iamsamgibbs 
> I've left one code comment where a validation change is preventing the manage permissions form from submitting with permissions.

> The completed view for "manage colony objective" is also missing the percentage sign after the number.

Fixed!

> (Unrelated to this issue, but I also noticed the "edit team" action is missing the description field, but it looks like the completed action view would show this correctly if it were there - not sure if that is quick and easy for you to add in with this PR, or can open as a new issue)

This was fixed when I did the rebase but yeah, the fix was gonna be easy and could have been included in this PR :)

**Review by iamsamgibbs — APPROVED** [review]
Works great now, thanks for the changes! 🎉

**Review by warrencolliver — APPROVED** [review]
working fine for me! great work

**Review by iamsamgibbs — APPROVED** [review]

**Review by bassgeta — APPROVED** [review]

**Pull request merged**